src/odemis/odemisd/modelgen.py

```diff
@@ -453,7 +461,15 @@ def _make_args(self, name):
                 else:
                     # the child has a class or is created by another component
                     # => we explicitly reuse it
-                    init["children"][internal_role] = self._get_component_by_name(child_name)
+                    init["dependencies"][internal_role] = self._get_component_by_name(child_name)
+        if "dependencies" in init and class_name == "Microscope":
+            raise SemanticError("Error in microscope file: "
+                                "microscope class '%s' should not have 'dependencies' argument." % name)
+        if "dependencies" in attr:
+            init["dependencies"] = {}
```

Comment:
I'm a little confused how the legacy depedencies are handled. IMHO, it'd be best if children which have a class (or a creator which is not the given component) are automatically moved to the "dependencies". However, from the code just above, it seems you're just handling them on-the-fly the old way. It should work too. However, if there is a mix legacy + new dependencies, the first init["dependencies"] (from the legacy) is going to be silently dropped. I'd suggest to either raise an Error, or merge them, and log a warning.